File: python/separate-the-digits-in-an-array.py
class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
        output = []
        for num in nums:
            buffer = []
            while 0 < num:
                buffer.append(num % 10)
                num //= 10
            while 0 < len(buffer):
                output.append(buffer.pop())
        return output

File: go/separate-the-digits-in-an-array.go
func separateDigits(nums []int) []int {
	output := make([]int, 0)
	for _, num := range nums {
		buffer := make([]int, 0)
		for 0 < num {
			buffer = append(buffer, num%10)
			num /= 10
		}
		for 0 < len(buffer) {
			i := len(buffer) - 1
			output = append(output, buffer[i])
			buffer = buffer[:i]
		}
	}
	return output
}
